CardGamesSprint4.py

```diff
@@ -0,0 +1,166 @@
+
+import random
+import os
+
+cardImages = []
+values = list(range(1,14))
+suits = ["Spades", "Clubs", "Hearts", "Diamonds"]
+
+def find_root_dir():
+  cwd = os.getcwd()
+  while 'source' not in os.listdir():
+    os.chdir('..')
+    cwd = os.path.join( cwd, '..')
+  return cwd
+
+class Card:
+  def __init__(self, suit, value, image, cardBack):
+    self.cardBack = cardBack
+    self.suit = suit
+    self.value = value
+    self.image = image
+    self.shortImage = []
+    for line in self.image:
+      self.shortImage.append(line[:4])
+
+  def __eq__(self, other):
+    if not type(other) == Card:
+      return False
+    return self.suit == other.suit and \
+      self.value == other.value
+
+class Deck:
+  def __init__(self):
+    root_dir = os.path.join( find_root_dir(), 'source')
+    cards_file = f'{root_dir}{os.path.sep}playing_cards.txt'
+    with open(cards_file, "r") as cards:
+      cardBack = []
+      for _ in range(6):
+        line = cards.readline()
+        cardBack.append(line.replace("\n",""))
+      card = []
+      level = 0
+      for line in cards.readlines():
+        if len(line) == 1:
+          cardImages.append(card)
+          level = 0
+          card = []
+          continue
+        card.append(line.replace("\n",""))
+        level += 1
+      cardImages.append(card)
+    
+    deck = []
+    index = 0
+    for suit in suits:
+      for value in values:
+        deck.append(Card(suit, value, cardImages[index], cardBack))
+        index += 1
+    
+    self.cards = deck
+    self.size = len(deck)
+    self.cardBack = cardBack
+    self.discarded = []
+
+  def reset(self):
+    self.cards += self.discarded
+    self.discarded = []
+    self.size = len(self.cards)
+
+  def shuffle(self):
+    random.shuffle(self.cards)
+
+  def getCard(self):
+    card = self.cards.pop()
+    self.size -= 1
+    self.discarded.append(card)
+    return card
+####################################################################################
+  def count_cards(deck):
+  #How many cards are in the deck assigned to "num_cards"
+  num_cards = len(deck)
+  #For each iteration the number of cars in the deck decrease by 1 card
+  for i in range(num_cards):
+    print("There are "+ str(num_cards - i) +" more cards in the deck")
+#when the deck reaches 0, then the game is over
+  if num_cards == 0:
+      print("The deck is now empty. Game over, Thanks for playing!")
+
+  def sayGoFish():
```

Comment:
Looks good. Function does the job. I'd say add some comments but it's simple enough to follow along.

Reply:
Sure, I think that's a great idea,